### PR TITLE
Filter products

### PR DESCRIPTION
**Pull request by VedanthB**
-  filter by category 
- filter by brand 
- filter by ratings

### CODE REVIEW

File: src/context/reducers/filterReducer.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { filtersConstants } from '../constants/filtersConstants'
+
+// sortBy: null,
```
Comment (Mohitkumar33) [inline src/context/reducers/filterReducer.js:3]: remove commented code

File: src/context/providers/FilterProvider.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        filteredByBrandProductsList,
+        filtersState.ratingsLessThan
+    )
+
+    let sortedFilteredList = sortProducts(
+        filteredByRatingsProductList,
+        filtersState.sortBy
+    )
+
+    console.log(sortedFilteredList)
```
Comment (Mohitkumar33) [inline src/context/providers/FilterProvider.js:60]: remove console

File: src/context/constants/Test.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
+import React, { useReducer } from "react";
+import "./styles.css";
+
+import faker from "faker";
+
+faker.seed(123);
+
+const data = [...Array(50)].map((item) => ({
```
Comment (Mohitkumar33) [inline src/context/constants/Test.js:8]: why you are using this faker data when you have data in you backend file saved

File: src/context/reducers/filterReducer.js (hunk 1 above)
Comment (Mohitkumar33) [inline src/context/reducers/filterReducer.js:3]: remove commented code

File: src/utils/filterProducts.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        filterProducts = [...filterProducts, ...filteredProductsByBrand]
+    }
+
+    return filterProducts
+}
+
+export const filterByRatings = (products, ratingsLessThan) => {
+    console.log(ratingsLessThan)
```
Comment (Hardiegogo) [inline src/utils/filterProducts.js:52]: You can remove this console.log.

File: src/context/reducers/filterReducer.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+//   skateboards: false,
+// },
+// brands: {
+//   vans: false,
+//   "urban-monkey": false,
+//   thrasher: false,
+//   "santa-cruz": false,
+// },
+// ratingsGreaterThan: 0,
+// priceLessThan: 20000,
```
Comment (Hardiegogo) [inline src/context/reducers/filterReducer.js:17]: Try removing unused code.

File: src/context/providers/FilterProvider.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+    let filteredByRatingsProductList = filterByRatings(
+        filteredByBrandProductsList,
+        filtersState.ratingsLessThan
+    )
+
+    let sortedFilteredList = sortProducts(
+        filteredByRatingsProductList,
+        filtersState.sortBy
+    )
```
Comment (Hardiegogo) [inline src/context/providers/FilterProvider.js:58]: Maybe you can try using compose function for this use -case. (Just a suggestion)

**Review by Hardiegogo — COMMENTED** [review]
I have suggested some minor change, you can have a look at them.
Other wise nicely structured code.

**Pull request merged**